src/lib/blkalloc/bitmap_blk_allocator.cpp

```diff
@@ -0,0 +1,201 @@
+/*********************************************************************************
+ * Modifications Copyright 2017-2019 eBay Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ *********************************************************************************/
+#include <homestore/homestore.hpp>
+#include <homestore/meta_service.hpp>
+#include <homestore/checkpoint/cp_mgr.hpp>
+#include "bitmap_blk_allocator.h"
+#include "meta/meta_sb.hpp"
+#include "common/homestore_utils.hpp"
+
+namespace homestore {
+BitmapBlkAllocator::BitmapBlkAllocator(BlkAllocConfig const& cfg, bool is_fresh, chunk_num_t id) :
+        BlkAllocator(cfg, id), m_blks_per_portion{cfg.m_blks_per_portion} {
+    if (is_persistent()) {
+        meta_service().register_handler(
+            get_name(),
+            [this](meta_blk* mblk, sisl::byte_view buf, size_t size) {
+                on_meta_blk_found(voidptr_cast(mblk), std::move(buf), size);
+            },
+            nullptr);
+    }
+
+    if (is_fresh || !is_persistent()) {
+        m_disk_bm = std::make_unique< sisl::Bitset >(m_num_blks, m_chunk_id, m_align_size);
```

Comment:
We are allocating memory for disk_bm when is_persistent return false, is it expected because we always access this bitmap on 1st cp? 
If so can we avoid doing that if it is not a persistent bitmap to save some memory budget which will not be used?

Reply:
Good catch. I think its a mistake to call this !is_persistent(). I think we need to create only if
`is_fresh && is_persistent`